### PR TITLE
WIP: Backwards compatibility for WebGL1 extensions

### PR DESCRIPTION
**Pull request by jtorresfabra**
Not for merge. Just to see the strategy for backwards compatibility.
Also some changes in order to compile without errors.

### CODE REVIEW

File: sources/osg/WebGLCaps.js [hunk 1]
```diff
@@ -363,6 +383,7 @@ WebGLCaps.prototype = {
             }
 
             ext[ sup ] = gl.getExtension( sup );
+            this.getAndApplyExtension( gl, sup );
```
Comment (cedricpinson) [inline sources/osg/WebGLCaps.js]: Does it really works for all extension ? If signature between extension and core of webgl2 it will not work

**Comment by cedricpinson** [comment]
what happens if on a page you have a webgl1 context and a webgl2, how webglcaps support that ?

**Comment by jtorresfabra** [comment]
Good question. I'm going to implement test example with this configuration to assure all works as expected. 

BTW: draw buffers also works with minor changes.

**Comment by jtorresfabra** [comment]
@cedricpinson just modified the multipleviewers example to have each viewer with a different webGL version. It fails, but it seems it is related to Uniforms, not to webGLCaps. that seems to work. I'll keep digging.

**Comment by cedricpinson** [comment]
Maybe we can also say that if you have webgl2 you just have to use two same webgl context. Not sure mixing is a real usecase

**Comment by jtorresfabra** [comment]
Ok I found the error:
https://github.com/cedricpinson/osgjs/blob/master/sources/osg/Uniform.js#L57

We cannot cache when different type of context comes in. Without cache it works with different contexts.

File: sources/osg/Uniform.js [hunk 2]
```diff
@@ -54,15 +54,25 @@ Uniform.prototype = {
 
     apply: function UniformApply( gl, location ) {
 
-        if ( !this._cache )
+        if ( !this._cache || this._cacheDirty === true ) {
             this._cache = gl[ this._glCall ];
-
+            this._dirtyCache = false;
```
Comment (stephomi) [inline sources/osg/Uniform.js]: this._cacheDirty = false;

Also why exposing gl call signature? I think it's supposed to be kept private

Comment (jtorresfabra) [inline sources/osg/Uniform.js]: > this._cacheDirty = false;
arg, well spotted!
> Also why exposing gl call signature? I think it's supposed to be kept private
I guess I did it for convenience so each time glSignature changes it dirties the cache.
https://github.com/jtorresfabra/osgjs/blob/653a94d5a8f160be5fa6aa2756ccc543325ec33b/sources/osg/Uniform.js#L169
 But I can keep it private. I'll fix it.

**Comment by jtorresfabra** [comment]
Closing this as https://github.com/cedricpinson/osgjs/pull/765 has been merged

**Comment by jtorresfabra** [comment]
Reopen the PR just to review it.

**Comment by jtorresfabra** [comment]
Closing it, does not make sense anymore.

**Pull request closed without merging**